include/arch/arc/v2/misc.h

```diff
@@ -27,6 +27,13 @@ static inline uint32_t arch_k_cycle_get_32(void)
 {
 	return sys_clock_cycle_get_32();
 }
+
+extern uint64_t sys_clock_cycle_get_64(void);
+
+static inline uint64_t arch_k_cycle_get_64(void)
+{
+	return sys_clock_cycle_get_64();
```

Comment:
Depending on the specific hardware configuration of a specific ARC CPU, there could be a 64-bit counter available.
We can add support for this later.
One question: if both 32-bit and 64-bit cycle counters are provided, are there any constraints or expectations on their relation? E.g. should the 32-bit one always be the least significant bits of the 64-bit one? Same frequency?

Reply:
Technically speaking, the lower 32 bits do not need to be the same, but they should have the same frequency

Reply:
I think it is best for the two to be synchronized and the lower 32 bits of the 64 bit timer should always be the same as the 32 bit timer.

Reply:
Pedantry: there's no way to express that requirement in a way that is actually implementable.  This isn't a synchronization problem, you can't "lock" against the passage of time.   At best we should just specify that the two APIs should ideally be backed by the same clock, or at worst have no practically detectable offset nor rate skew.

Reply:
I did not intend this to be stopping the clock between different reads... I am saying if you do the following that it will be as expected from a single timer.

1. first_read = read 64 bit timer
2. second_read = read 32 bit timer
3. third_read = read 64 bit timer

(first_read & 0xFFFFFFFF) <= second_read <= (third_read & 0xFFFFFFFF)

Reply:
Agreed - my main point was indeed that the requirement or expected behavior is somewhere specified and documented. I guess there are pro's and con's for both approaches, but since I would not expect both to be used and combined (comparing 32-bit one with 64 bit one or so), I'd say only have the same frequency.